### PR TITLE
fix: update offchain status

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

fix bug: can't update offchain status to 0

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 
* fix bug: can't update offchain status to 0
* support OffChainStatusSpAsDelegatedAgentDisabled

### Potential Impacts
* QA

### CODE REVIEW

**Review by constwz — APPROVED** [review]

File: modular/blocksyncer/database/bucket.go [hunk 1]
```diff
@@ -40,6 +40,19 @@ func (db *DB) UpdateBucketByNameToSQL(ctx context.Context, bucket *models.Bucket
 	return stat.SQL.String(), stat.Vars
 }
 
+func (db *DB) UpdateBucketOffChainStatus(ctx context.Context, bucket *models.Bucket) (string, []interface{}) {
+	stat := db.Db.Session(&gorm.Session{DryRun: true}).Table((&models.Bucket{}).TableName()).
+		Where("bucket_name = ?", bucket.BucketName).
+		Select("off_chain_status", "update_at", "update_tx_hash", "update_time").
+		Updates(models.Bucket{
```
Comment (mliu233) [inline modular/blocksyncer/database/bucket.go:47]: use Update(bucket) would be enough I think?

File: modular/blocksyncer/modules/bucket/bucket_handle.go [hunk 2]
```diff
@@ -16,25 +18,28 @@ import (
...
+	EventDiscontinueBucket:        true,
+	EventCompleteMigrationBucket:  true,
+	EventToggleSPAsDelegatedAgent: true,
 }
 
 type OffChainStatus int
 
 const (
 	OffChainStatusIsLimited OffChainStatus = 1 << iota // 1
+	OffChainStatusSpAsDelegatedAgentDisabled
```
Comment (mliu233) [inline modular/blocksyncer/modules/bucket/bucket_handle.go:42]: would be better to write a comment like // 1 << 1 which is 00000010, just so might be better to understand later.

File: modular/blocksyncer/database/bucket.go (hunk 1 above)
Comment (BarryTong65) [reply modular/blocksyncer/database/bucket.go:47]: According to the official Gorm documentation, "Updates" is used to update multiple columns, while "Update" is generally used to update a single one.
image

File: modular/blocksyncer/modules/bucket/bucket_handle.go (hunk 2 above)
Comment (BarryTong65) [reply modular/blocksyncer/modules/bucket/bucket_handle.go]: fixed

File: modular/blocksyncer/database/bucket.go (hunk 1 above)
Comment (BarryTong65) [reply modular/blocksyncer/database/bucket.go]: fixed

**Review by annielz — APPROVED** [review]

**Review by ruojunm — APPROVED** [review]

**Pull request merged**